grails-app/domain/org/pih/warehouse/product/Product.groovy

```diff
@@ -453,6 +454,42 @@ class Product implements Comparable, Serializable {
         }
     }
 
+    InventoryItem getDefaultInventoryItem() {
+        return getInventoryItem(Constants.DEFAULT_LOT_NUMBER)
+    }
+
+
+    InventoryItem getInventoryItem(String lotNumber) {
+        return getInventoryItem(lotNumber, null)
+    }
+
+    // FIXME we could also just traverse the inventoryItems association rather than making an extra query
+    // FIXME but i wanted to move towards a detached criteria / named query fetch
+    // FIXME the expiration date should be involved here, but we don't require it in other places
+    //  so i'll leave that for as a problem for future me
+    InventoryItem getInventoryItem(String lotNumber, Date expirationDate) {
+
+        // Treat the default lot number (DEFAULT) as null
+        if (Constants.DEFAULT_LOT_NUMBER.equalsIgnoreCase(lotNumber)) {
+            lotNumber = null
+        }
+
+        InventoryItem.createCriteria().get {
+            and {
+                eq("product", this)
+                if (lotNumber) {
+                    eq("lotNumber", lotNumber)
+                } else {
+                    or {
+                        isNull("lotNumber")
+                        eq("lotNumber", "")
+                    }
+                }
+            }
+        } as InventoryItem
```

Comment:
It can be only my own preference, but I prefer to use the `return` keyword

Reply:
Yeah, same.  It was there but i was getting a weird IDE error so while I was trying to fix that I think I accidentally removed the "return"
![image](https://github.com/openboxes/openboxes/assets/1162935/ba67ca07-f737-45ee-a804-ecb3a79a04d6)

Reply:
Probably adding `as InventoryItem` behind the closure should solve the issue, but I am not sure.

Reply:
Yes it does. The `as InventoryItem` doesn't feel right either though. But I was just saying that i tried a few things (including removing the return) so that's why it was in that state. Should be resolved now.